sklearn/preprocessing/_data.py

```diff
@@ -3236,14 +3236,21 @@ def _yeo_johnson_optimize(self, x):
 
         Like for Box-Cox, MLE is done via the brent optimizer.
         """
+        x_tiny = np.finfo(x.dtype).tiny
```

Comment:
np.var uses float64 accumulator. I think it's more appropriate to always use float64's tiny.